source/fundamentals/bson.txt

```diff
@@ -0,0 +1,301 @@
+=================
+Working with BSON
+=================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about how the Go Driver handles conversions
+between BSON and Go objects. The process of converting a Go object to
+BSON is called **marshalling**, while the reverse process is called **unmarshalling**. 
+
+You should read this guide if you want to learn more about how the Go Driver
+represents BSON data or need to adjust default marshalling
+and unmarshalling behaviors.
+
+Data Types
+----------
+
+MongoDB stores documents in a binary representation called :manual:`BSON
+</reference/bson-types/>` that allows for easy and flexible data processing. The Go
+Driver represents BSON data using two families of types: ``D`` and ``Raw``.
+
+The ``D`` family contains four types:
+
+- ``D``: An ordered representation of a BSON document (slice)
+- ``M``: An unordered representation of a BSON document (map)
+- ``A``: An ordered representation of a BSON array
+- ``E``: A single element inside a D type
+
+Of these, you can use the ``D`` and ``M`` types to
+build representations of BSON using native Go types. 
+
+The following example demonstrates how to construct a query filter using the
+``bson.D`` type to match documents with a ``quantity`` field value greater
+than 100:
+
+.. code-block:: go
+   
+   filter := bson.D{{"quantity", bson.D{{"$gt", 100}}}}
+
+You can use ``Raw`` types to validate and retrieve elements from a slice
+of bytes, which is useful if you don't want to unmarshal BSON into
+another type. 
+
+In Go, a **struct** is a collection of data fields with declared data
+types. The Go Driver can marshal/unmarshal structs and other native Go
+types to/from BSON using a `configurable codec system <https://pkg.go.dev/go.mongodb.org/mongo-driver/bson/bsoncodec>`_.
```

Comment:
It's kinda weird to go straight into talking about structs, and then have a separate section for _Struct Tags_. Consider renaming the _Struct Tags_ section to something like _BSON as a Go `struct`_ and move this paragraph into the beginning of that section.

Reply:
I moved the sentence but kept the subheading of the section as Struct Tags because I think that's helpful for users navigating the "On this page" table of contents

Reply:
Ok, seems reasonable.